packages/codegen/src/generator/ErrorsGen.ts

```diff
@@ -0,0 +1,45 @@
+import { ApiGen } from './ApiGen';
+import { stringLowerFirst } from '@polkadot/util';
+import { beautifySourceCode, commentBlock, compileTemplate } from './utils';
+
+export class ErrorsGen extends ApiGen {
+  generate() {
+    const { pallets } = this.metadata;
+
+    this.typesGen.clearCache();
+    this.typesGen.typeImports.addKnownType('GenericChainErrors', 'GenericModuleError');
+
+    let defTypeOut = '';
+    for (let pallet of pallets) {
+      const error = pallet.error;
+      if (!error) {
+        continue;
+      }
+
+      const errors = this.#getErrorVariants(error);
+
+      defTypeOut += `${stringLowerFirst(pallet.name)}: {
+        ${errors.map(({ name, docs }) => `${commentBlock(docs)}${name}: GenericModuleError`).join(',\n')}
+      },`;
+    }
+
+    const importTypes = this.typesGen.typeImports.toImports();
+    const template = compileTemplate('errors.hbs');
+
+    return beautifySourceCode(template({ importTypes, defTypeOut }));
+  }
+
+  #getErrorVariants(errorId: number) {
+    const def = this.metadata.types[errorId];
+    if (!def) {
+      throw new Error(`Error def not found: ${JSON.stringify(def)}`);
+    }
```

Comment:
Let's use assert for this type of checking, and is `def` is `undefined` there is no need to use `${JSON.stringify(def)}`.
```suggestion
    import { assert } from '@delightfuldot/utils';
    assert(def, `Error def not found for id ${errorTypeId}`);
```